src/components/material.js

```diff
@@ -55,31 +62,46 @@ module.exports.Component = registerComponent('material', {
     if (!schema) { error('Unknown shader schema ' + shader); }
     if (currentShader && newShader === currentShader) { return; }
     this.extendSchema(schema);
-    this.updateBehavior();
+    this.updateTick();
   },
 
-  updateBehavior: function () {
+  /**
+   * Update tick handler that continuously updates certain shader uniforms if the shader
+   * specifies certain uniform types like `time`.
+   */
+  updateTick: function () {
```

Comment:
I have a different way to handle this on my shader toy PR. Instead of being hard coded the schema of the property defines an update function if it's meant to be updated on every frame.